Uncomment and fix CodeGeneration tests

svick opened the pull request:
I don't know why these tests were commented out using `#if false`, but I think they should be either uncommented or deleted.

What I did:

1. Removed the `#if`s.
2. Fixed the tests.
3. Added `[Trait(Traits.Feature, Traits.Features.CodeGeneration)]`.
4. Added `Skip` to tests that I didn't know how to fix.

Regarding 4., several tests in [`ExpressionGenerationTests`](https://github.com/dotnet/roslyn/blob/7dfc2fa/src/EditorFeatures/Test/CodeGeneration/ExpressionGenerationTests.cs) (and I assume all tests in [`ExpressionPrecedenceGenerationTests`](https://github.com/dotnet/roslyn/blob/7dfc2fa/src/EditorFeatures/Test/CodeGeneration/ExpressionPrecedenceGenerationTests.cs)) fail, because the code generator creates unnecessary parentheses. I'm not sure what is the right way to test those, so I have added `Skip = "parens"` to those tests (and didn't uncomment `ExpressionPrecedenceGenerationTests`).

---

In case `master` is not the right branch for this kind of PR at the moment, feel free to retarget it to a different branch.

sharwell commented on src/EditorFeatures/Test/CodeGeneration/ExpressionGenerationTests.cs:
```diff
@@ -1,436 +1,434 @@
         }
 
-        [WpfFact]
+        [Fact(Skip = "parens")]
```
❗️ Create an issue to address the remaining items after this is merged. Then replace `parens` with a link to the issue. (You can use one issue to cover all the failing tests that are skipped in this PR.)

sharwell commented:
@svick Since this took so long to get to, let me know if you'd like me to go ahead and make the necessary change myself so this can get merged. 😄

svick commented:
@sharwell I have rebased this PR on top of master, updated it, uncommented `ExpressionPrecedenceGenerationTests` and linked the skipped tests to the newly created issue https://github.com/dotnet/roslyn/issues/26586.

I think this PR should be ready now.

sharwell commented:
Merging this test-only change 👍 

Thanks @svick !
